src/charm.py

```diff
@@ -301,12 +301,20 @@ def _configure(self, _):
 
         self.remote_write_provider.update_endpoint()
         self.grafana_source_provider.update_source(self.external_url)
-        self.unit.set_workload_version(self._prometheus_server.version())
         self.unit.status = ActiveStatus()
 
-    def _update_status(self, event):
-        """Fired intermittently by the Juju agent."""
-        self.unit.set_workload_version(self._prometheus_server.version())
+    def _on_pebble_ready(self, event) -> None:
+        """Pebble ready hook.
+
+        This runs after the the workload container starts.
+        """
+        self._configure(event)
+        if version := self._prometheus_version:
```

Comment:
Walrus again

Reply:
Pretty sure this code only ever runs on modern python.